### PR TITLE
Tutorial on working with SBOL Data Model

### PR DESCRIPTION
**Pull request by PiotrZakrzewski**
Based on https://github.com/SynBioDex/SBOL-Notebooks/blob/main/iwbda_2023_examples.ipynb

The intention is to provide a set of concrete examples on how to create and modify different SBOL data structures.

### CODE REVIEW

**Comment by PiotrZakrzewski** [comment]
@jakebeal Could you help me with this draft? Are there any stylistic conventions I should adapt to? What content do you think this tutorial should have at the minimum?

**Comment by jakebeal** [comment]
If you're willing, it would be great to start by just mapping the whole thing over, as is. The style that you've got is looking good in the test build: https://pysbol3--449.org.readthedocs.build/en/449/sbol_data_model.html

With regards to the `Range` content that you are contributing, it would be nice to have it be actually executable in combination with the rest of the code in the document. For example, you might use the `Range` to make a `SequenceFeature` in `e0040 ` that marks the start codon of its `e0040_sequence` 

Finally: don't worry about the build error that's there right now -- that is a different issue that showed up on the main branch too, due to a chance in pylint standards, and I've just filed issue #450 for dealing with that.

**Comment by PiotrZakrzewski** [comment]
@jakebeal I pushed the draft of mapping the Jupyter tutorial as-is, without the Range part. Once this is merged I will make another PR just for the Range part.

I will fix some small formatting issues and proof read the draft tomorrow before I remove the Draft label, but feel free to leave some remarks already.

**Comment by PiotrZakrzewski** [comment]
@jakebeal ready

**Comment by tcmitchell** [comment]
@PiotrZakrzewski  thank you, this looks great to me. @jakebeal will need to review it as well. Thanks very much for the contribution!

**Review by tcmitchell — APPROVED** [review]

**Comment by tcmitchell** [comment]
@jakebeal it would be nice to have you review the document before merging. If you think it's fine without review please add a comment and I'll go ahead and merge. Thanks!

File: docs/sbol_data_model.rst [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+This documentation is based on a Jupyter Notebook tutorial presented at IWBDA 2023, demonstrating the SBOL v3 data model.
+
+`IWBDA 2023 SBOL 3 Tutorial Slides <https://github.com/SynBioDex/Community-Media/blob/master/2023/IWBDA23/SBOL3-IWBDA-2023.pptx>`_
+
+`Jupyter Notebook <https://github.com/SynBioDex/SBOL-Notebooks/blob/main/iwbda_2023_examples.ipynb>`_
+
+Installation
+=============
+
+Sbol Utilities is a Python package that provides a set of utility functions for working with the SBOL3 data model.
```
Comment (jakebeal) [inline docs/sbol_data_model.rst:17]:
```suggestion
SBOL Utilities is a Python package that provides a set of utility functions for working with the SBOL3 data model. 
```

File: docs/sbol_data_model.rst [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+
+.. code-block:: python
+
+    from sbol3 import *
+    from sbol_utilities.calculate_sequences import compute_sequence
+    from sbol_utilities.component import *
+    from sbol_utilities.helper_functions import url_to_identity
+    import tyto
+
+We will use `igem` suffix as the default namespace for the examples in this tutorial.
```
Comment (jakebeal) [inline docs/sbol_data_model.rst:39]:
```suggestion
We will use an `igem` suffix as the default namespace for the examples in this tutorial.
```

**Review by jakebeal — APPROVED** [review]
Apologies for the delay in review. I have finally had a chance to go carefully end-to-end. I found a couple of typos, but other than that it looks goof to go!

**Pull request merged**